src/compiler/binder.ts

```diff
@@ -2791,14 +2794,18 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
             parent = saveParent;
         }
         inStrictMode = saveInStrictMode;
+        currentFlowJsDoc = saveCurrentFlowJsDoc;
     }
 
     function bindJSDoc(node: Node) {
         if (hasJSDocNodes(node)) {
             if (isInJSFile(node)) {
+                const saveCurrentFlow = currentFlow;
+                currentFlow = currentFlowJsDoc;

```

Comment:
I have also considered moving `bindJSDoc` outside of `bindChildren` and calling it earlier - before `bindContainer` and some other function. This seemed like more work and *maybe* there is a reason why it shouldn't be done - I hope that a reviewer might have some intuition for this 😅